[CI] RetrySearchIntegTests testRetryPointInTime failing

**Build Scans:**
- [elasticsearch-periodic #4976 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/wqvu4uwg44vok)
- [elasticsearch-pull-request #42500 / part-2](https://gradle-enterprise.elastic.co/s/uqwl7kkcvj2le)
- [elasticsearch-pull-request #42454 / part-2](https://gradle-enterprise.elastic.co/s/hyvdzlaqnkp5y)
- [elasticsearch-periodic #4964 / openjdk21_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/np4y5utjpmwn6)
- [elasticsearch-pull-request #42364 / part-2](https://gradle-enterprise.elastic.co/s/26dz7wiykv3li)
- [elasticsearch-intake #13137 / part2](https://gradle-enterprise.elastic.co/s/xatlrtg3mdwqc)
- [elasticsearch-pull-request #41670 / part-2](https://gradle-enterprise.elastic.co/s/4m75vrbiu7sda)
- [elasticsearch-pull-request #41627 / part-2](https://gradle-enterprise.elastic.co/s/kucijrb26i7uq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:searchable-snapshots:internalClusterTest" --tests "org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests.testRetryPointInTime" -Dtests.seed=ED5FE79C590CEF47 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=fr-DJ -Dtests.timezone=Pacific/Galapagos -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testRetryPointInTime),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Index{id='kohnffzyet/xEqwMTroQbSyvN_jPBoIOw/2N66dIAWS1ms7kHiO2O0kg/4/_0.cfe/@0', seqNo=5, primaryTerm=1, version=2, autoGeneratedIdTimestamp=-1}
Expected: <3L>
     but: was <1L>
```

**Issue Reasons:**
- [main] 8 failures in test testRetryPointInTime (1.0% fail rate in 812 executions)
- [main] 5 failures in step part-2 (1.4% fail rate in 361 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (15.4% fail rate in 13 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (1.4% fail rate in 369 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -157,9 +157,6 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=snapshot/10_basic/Create a source only snapshot and then restore it}
   issue: https://github.com/elastic/elasticsearch/issues/117295
-- class: org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests
-  method: testRetryPointInTime
-  issue: https://github.com/elastic/elasticsearch/issues/117116
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testInferDeploysDefaultElser
   issue: https://github.com/elastic/elasticsearch/issues/114913

--- a/test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java
@@ -128,6 +128,7 @@ public abstract class AbstractSnapshotIntegTestCase extends ESIntegTestCase {
 
     @After
     public void assertConsistentHistoryInLuceneIndex() throws Exception {
+        internalCluster().beforeIndexDeletion();
         internalCluster().assertConsistentHistoryBetweenTranslogAndLuceneIndex();
     }
 

